core/arch/arm/plat-marvell/main.c

```diff
@@ -108,3 +114,36 @@ void console_init(void)
 #endif
 	register_serial_console(&console_data.chip);
 }
+
+#ifdef CFG_HW_UNQ_KEY_SUPPORT
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	uint8_t hw_unq_key[sizeof(hwkey->data)] __aligned(64);
+	uint64_t huk1 = 0, huk2 = 0;
+	vaddr_t fuse_huk1 = 0, fuse_huk2 = 0;
```

Comment:
- One variable per line please.
- How about making `fuse_huk1` and `fuse_huk2` `void *`? It seems you could get rid of the casts.

Reply:
> * One variable per line please.
Okay agree. Will make the change.
> * How about making `fuse_huk1` and `fuse_huk2` `void *`? It seems you could get rid of the casts.
correct I thought about it, but just for the consistency of type vaddr_t as per most of the places of optee os I forced to maintain the same.

Reply:
> * One variable per line please.
> * How about making `fuse_huk1` and `fuse_huk2` `void *`? It seems you could get rid of the casts.

- Okay agree. Will make the change.
- Correct I thought about it, but just for the consistency of type vaddr_t as per most of the places of optee os I forced to maintain the same.

Reply:
>     * Correct I thought about it, but just for the consistency of type vaddr_t as per most of the places of optee os I forced to maintain the same.

Because very often `phys_to_virt()` is used to obtain a base address, which is later used with byte offsets (base + x). `void *` would not work in this case and `vaddr_t` is appropriate. Here it is not the case and `void *` is definitely the right choice.

Reply:
>     * Correct I thought about it, but just for the consistency of type vaddr_t as per most of the places of optee os I forced to maintain the same.

Because very often `phys_to_virt()` is used to obtain a base address, which is later used with byte offsets (base + x). `void *` would not work in this case and `vaddr_t` is appropriate. Here it is not the case and `void *` is definitely the right choice.

Reply:
> > ```
> > * Correct I thought about it, but just for the consistency of type vaddr_t as per most of the places of optee os I forced to maintain the same.
> > ```
> 
> Because very often `phys_to_virt()` is used to obtain a base address, which is later used with byte offsets (base + x). `void *` would not work in this case and `vaddr_t` is appropriate. Here it is not the case and `void *` is definitely the right choice.

Agree, will make the change.